src/normalize/pathoverlay.ts

```diff
@@ -36,7 +37,11 @@ function getPointOverlay(
   markConfig: LineConfig<ExprRef | SignalRef> = {},
   encoding: Encoding<string>
 ): MarkConfig<ExprRef | SignalRef> {
-  if (markDef.point === 'transparent') {
+  if (
+    markDef.point === 'transparent' ||
+    // if the main mark is a single line/trail/area chart, create an invisible point overlay for label.
+    (!markDef.point && encoding.label && pathGroupingFields(markDef.type, encoding).length <= 0)
```

Comment:
- [ ] Probably need to update the condition here to not always does this.
- [ ] It feels hacky that the main compiler needs to depends on a mechanism inside normalize.